src/components/expenses/expense-listing/list-table.tsx

```diff
@@ -18,116 +18,119 @@ import {
+    await getExpense().then((res) => {
+      setTablelist(res.expenses);
+      setloading(false);
+    });
 
   const table = useReactTable({
-    data: expenses,
+    data: tablelist,
     columns: expenseListColumns,
     getCoreRowModel: getCoreRowModel(),
     onSortingChange: setSorting,
     getSortedRowModel: getSortedRowModel(),
+    onColumnVisibilityChange: setColumnVisibility,
     state: {
       sorting,
+      columnVisibility,
     },
   });
+  if (loading) return <p>Loading...</p>;
 
   return (
     <div>
-      <div className="mb-[8px] max-w-max ml-auto">
-        <DropdownMenu>
-          <DropdownMenuTrigger asChild>
-            <Button
-              variant="outline"
-              className="rounded-md font-normal"
+      <div className='mb-[8px] space-x-[12px] max-w-max ml-auto'>
+        <AddNewExpenseDialog />
+        {!!tablelist.length && (
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button variant='outline' className='rounded-[10px] font-normal'>
+                Toggle columns{" "}
+                <GoChevronDown className='ml-[8px] h-[16px] w-[16px]' />
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent
+              align='end'
+              className='w-dropdown-trigger-width'
             >
-              Toggle columns <GoChevronDown className="ml-[8px] h-[16px] w-[16px]" />
-            </Button>
-          </DropdownMenuTrigger>
-          <DropdownMenuContent
-            align="end"
-            className="w-dropdown-trigger-width"
-          >
-            {table
-              .getAllColumns()
-              .filter((column) => column.getCanHide())
-              .map((column) => {
-                return (
-                  <DropdownMenuCheckboxItem
-                    key={column.id}
-                    checked={column.getIsVisible()}
-                    onCheckedChange={(value) =>
-                      column.toggleVisibility(!!value)
-                    }
-                  >
-                    {column.id}
-                  </DropdownMenuCheckboxItem>
-                );
-              })}
-          </DropdownMenuContent>
-        </DropdownMenu>
+              {table
+                .getAllColumns()
+                .filter((column) => column.getCanHide())
+                .map((column) => {
+                  return (
+                    <DropdownMenuCheckboxItem
+                      key={column.id}
+                      checked={column.getIsVisible()}
+                      onCheckedChange={(value) =>
+                        column.toggleVisibility(!!value)
+                      }
+                    >
+                      {column.id}
+                    </DropdownMenuCheckboxItem>
+                  );
+                })}
+            </DropdownMenuContent>
+          </DropdownMenu>
+        )}
       </div>
-      <DataTable table={table} />
+      {!!tablelist.length ? (
```

Comment:
No need to `!!` if using the value as a condition check (both ternary and `if`), but use it if using it as a value itself and you wanna convert it to a `boolean`. 
Eg.:
```ts
// no need
tableList.length ? ... : ...

// no need
if (tableList.length)

// needed
const isEmpty = !!tableList.length;

// needed
!!tableList.length && <div ... 
```
```